### PR TITLE
Fix recursive commands.

### PR DESCRIPTION
**Pull request by JBYoshi**
Fixes SpongePowered/SpongeForge#381.

### CODE REVIEW

File: src/main/java/org/spongepowered/common/command/MinecraftCommandWrapper.java [hunk 1]
```diff
@@ -209,8 +218,14 @@ public Text getUsage(CommandSource source) {
     }
 
     public static void setError(Throwable error) {
-        if (commandErrors.get() == noError) {
-            commandErrors.set(error);
+        Preconditions.checkNotNull(error);
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/command/MinecraftCommandWrapper.java]: Can static import this method.

Comment (JBYoshi) [inline src/main/java/org/spongepowered/common/command/MinecraftCommandWrapper.java]: Done.

File: src/main/java/org/spongepowered/common/command/MinecraftCommandWrapper.java [hunk 2]
```diff
@@ -61,7 +65,13 @@
     private static final String TRANSLATION_NO_PERMISSION = "commands.generic.permission";
     private final PluginContainer owner;
     protected final ICommand command;
-    private static final ThreadLocal<Throwable> commandErrors = new ThreadLocal<>();
+    private static final ThreadLocal<Deque<Throwable>> commandErrors = new ThreadLocal<Deque<Throwable>>() {
+
+        @Override
+        protected Deque<Throwable> initialValue() {
+            return new LinkedList<Throwable>();
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/command/MinecraftCommandWrapper.java]: can't you use a lambda here? We're not in a mixin.

**Comment by JBYoshi** [comment]
Okay, hold on... something's still going wrong. Don't merge yet.

File: src/main/java/org/spongepowered/common/command/MinecraftCommandWrapper.java (hunk 2 above)
Comment (JBYoshi) [inline src/main/java/org/spongepowered/common/command/MinecraftCommandWrapper.java]: No. `ThreadLocal` is a class, not a functional interface.

**Comment by JBYoshi** [comment]
@gabizou Fixed. Ready for merge.

**Comment by kenzierocks** [comment]
Yes you can use a lambda. ThreadLocal has a static method for this.

**Comment by JBYoshi** [comment]
@kenzierocks I'll take care of that tomorrow.

**Pull request closed without merging**